rfcs/text/0000-dss-deletion-daemon.md

```diff
@@ -0,0 +1,268 @@
+   not immediately affected by the deletion and remain accessible. The bundle and associated files are added to 
+   a deletion queue for regularly schedule **Physical Deletion**.
+   
+   A notification is sent out to subscribers of datastore deletions when a new bundle tombstone is created, so they can 
+   update their index appropriately.
+
+4) On a regular schedule the deletion queue is processed by code running within a secure boundary. An administrator 
+   may manually invoke the deletion process or wait for it run at it's scheduled time once daily.
+   The deletion daemon is responsible for performing **Physical Deletes** of data from the deletion queue. 
+   A physical delete makes all files associated with a bundle or bundle version inaccessible. This will make HEAD and 
+   GET requests against those file versions return a 404.
+   
+   A log entry is produced when ever the deletion daemon is run. It contains information on what bundles and files
+   have been logically deleted, the reason, the admin who executed the deletion, and info about the deletion markers. 
+   The same information is stored in the **Deletion Table**. The deletion table tracks all bundles and files currently
+   pending permanent deletion. The Deletion Table contains enough information to reverse Physical Deletes before 
+   the data is Permanently deleted.
+
+5) Data that has been physical deleted remain in the bucket for 7 days before being permanently deleted from the bucket.
+   This grace period allows the deletion to be reversed within that time period. After the grace period the data will 
+   be permanently deleted.
+   
+### Primary indexes
+
+Bundle tombstones are indexed verbatim by the Data Store. The JSON schema
+for deletion markers is defined such that there are no conflicts with the
+top-level keys in the index documents for regular bundles (`files`, `manifest`,
+`uuid`, `version` etc). Bundle tombstones are therefore compatible with
+regular bundle documents with respect to in the same index even if
+Elasticsearch's dynamic mapping is enabled.
+
+### Replication
+
+When a bundle is deleted it is removed from all replicas.
+
+### Consistency
+
+It is the administrator's responsibility to enumerate all bundles referring to
+a file that is to be deleted and issue a delete requests for each of
+them. If the administrator misses such a bundle (a *dangling bundle*), users
+requesting `GET /bundle` for a dangling bundle will be able to retrieve its
+manifest. Likewise, `POST /search` requests will return the metadata for
+dangling bundles. Secondary indexes like the Data Browser (Orange Box) will
+continue to hand out (meta)data for dangling bundles. If users request `GET
+/files` for a file contained in a dangling bundle and the requested file was
+referenced by another already logically (physically) deleted bundle, the Data
+Store will respond with a 404 (500) status code.
```

Comment:
Is there currently a way to list all bundles that a file is part of in the API? If not, we should consider adding one.

Reply:
We can accomplish this with an ES query of the manifest.